Lib/test/libregrtest/main.py

```diff
@@ -109,7 +109,7 @@ def __init__(self, ns: Namespace, _add_python_opts: bool = False):
         self.random_seed: int =  (
             ns.random_seed
             if ns.random_seed is not None
-            else random.randrange(100_000_000)
+            else random.Random().getrandbits(32)
```

Comment:
What's the point of creating a new Random() instead just for a function call? random module already has one at the module level. Just use:

```suggestion
            else random.getrandbits(32)
```

Ah, maybe the function wasn't available at the module level before. But it tested that it's accessible on Python >= 3.11.

Reply:
This approach guarantees that our seed generation is not affected by the global random instance which we seed later. It is just 8 chars to make it safer and add more isolation. `pytest-randomly` does the same.

Reply:
> is not affected by the global random instance which we seed later

Would you mind to elaborate? Isolated from what issues or events?

Reply:
We modify global `random` instance in tests, so, it *theoretically* can affect `.getrandbits` result. I don't think that there are any practical ways to do this *at the moment*, but you never know :)

Anyway, I don't have any strong feelings about this line of code. Feel free to press "commit suggestion" if you prefer the shorter version! 👍

Reply:
> We modify global random instance in tests

Regrtest constructor is called *before* any test is run.

If a test overrides a stdlib function, it *must* restores the stdlib in its previous state when the test completes. Well, save_env.py is incomplete and bugs are possible. But for this very specific line, it doesn't matter ;-)